rules/camunda-cloud-1-0-checks.js

```diff
@@ -29,69 +39,99 @@ module.exports = [
   'bpmn:TextAnnotation',
   {
     type: 'bpmn:BoundaryEvent',
-    check: hasEventDefinitionOfType([
-      'bpmn:ErrorEventDefinition',
-      'bpmn:MessageEventDefinition',
-      'bpmn:TimerEventDefinition'
-    ])
+    check: checkEvery(
+      hasEventDefinitionOfType([
+        'bpmn:ErrorEventDefinition',
+        'bpmn:MessageEventDefinition',
+        'bpmn:TimerEventDefinition'
+      ]),
+      checkIf(
+        checkEventDefinition(hasErrorReference),
+        hasEventDefinitionOfType('bpmn:ErrorEventDefinition')
+      ),
+      checkIf(
+        checkEventDefinition(hasZeebeSubscription),
+        hasEventDefinitionOfType('bpmn:MessageEventDefinition')
+      )
+    )
+  },
+  {
+    type: 'bpmn:CallActivity',
+    check: checkEvery(
+      hasZeebeCalledElement,
+      hasZeebeLoopCharacteristics
+    )
   },
   {
     type: 'bpmn:EndEvent',
-    check: checkSome(
-      hasNoEventDefinition,
-      hasEventDefinitionOfType([
-        'bpmn:ErrorEventDefinition'
-      ])
+    check: checkEvery(
+      hasEventDefinitionOfTypeOrNone('bpmn:ErrorEventDefinition'),
+      checkIf(
+        checkEventDefinition(hasErrorReference),
+        hasEventDefinitionOfType('bpmn:ErrorEventDefinition')
+      )
     )
   },
   {
     type: 'bpmn:IntermediateCatchEvent',
-    check: hasEventDefinitionOfType([
-      'bpmn:MessageEventDefinition',
-      'bpmn:TimerEventDefinition'
-    ])
+    check: checkEvery(
+      hasEventDefinitionOfType([
+        'bpmn:TimerEventDefinition',
+        'bpmn:MessageEventDefinition'
+      ]),
+      checkIf(
+        checkEventDefinition(hasZeebeSubscription),
+        hasEventDefinitionOfType('bpmn:MessageEventDefinition')
+      )
+    )
   },
   {
     type: 'bpmn:Process',
     check: hasNoLanes
   },
   {
     type: 'bpmn:ReceiveTask',
-    check: checkSome(
-      hasNoLoopCharacteristics,
-      hasLoopCharacteristicsOfType('bpmn:MultiInstanceLoopCharacteristics')
+    check: checkEvery(
+      checkFlowNode(hasZeebeSubscription),
+      hasZeebeLoopCharacteristics
     )
   },
   {
     type: 'bpmn:ServiceTask',
-    check: checkSome(
-      hasNoLoopCharacteristics,
-      hasLoopCharacteristicsOfType('bpmn:MultiInstanceLoopCharacteristics')
+    check: withTranslations(
+      checkEvery(
+        hasZeebeLoopCharacteristics,
+        hasZeebeTaskDefinition
+      ),
+      {
+        'Element of type <bpmn:ServiceTask> must have <zeebe:TaskDefinition> extension element': 'Must have implementation'
```

Comment:
I'm confused here. When I create a ServiceTask, I get "Must have implementation", which isn't very specific. Is this a placeholder for the friendly messages?

Reply:
This is just an example to verify that we can easily adjust the messages once we've come up with friendly error messages. So this is going to change. 😉

Reply:
Ok makes sense